### PR TITLE
GH-44656: [GLib] Add GArrowBinaryViewDataType

### PR DESCRIPTION
**Pull request by hiroyuki-sato**
### Rationale for this change

The `arrow::BinaryViewType` has been introduced.
GLib needs to be implemented as the `GArrowBinaryViewDataType`.

### What changes are included in this PR?

Implement `GArrowBinaryViewDataType`.

### Are these changes tested?

YES

### Are there any user-facing changes?

NO

* GitHub Issue: #44656

### CODE REVIEW

File: c_glib/arrow-glib/basic-data-type.cpp [hunk 1]
```diff
@@ -69,6 +69,8 @@ G_BEGIN_DECLS
  *
  * #GArrowBinaryDataType is a class for the binary data type.
  *
+ * #GArrowBinaryViewDataType is a class for the binary data type.
```
Comment (kou) [inline c_glib/arrow-glib/basic-data-type.cpp:72]:
```suggestion
 * #GArrowBinaryViewDataType is a class for the binary view data type.
```
Anyway, could you move this to https://github.com/apache/arrow/blob/40b2fca4742e2692a917755fd8db2939e10fa02d/c_glib/arrow-glib/basic-data-type.cpp#L130 ?

File: c_glib/arrow-glib/basic-data-type.cpp [hunk 2]
```diff
@@ -813,6 +815,35 @@ garrow_binary_data_type_new(void)
...
+/**
+ * garrow_binary_view_data_type_new:
+ *
+ * Returns: The newly created binary view data type.
+ */
+GArrowBinaryViewDataType *
+garrow_binary_view_data_type_new(void)
+{
+  auto arrow_data_type = arrow::binary_view();
+
```
Comment (kou) [inline c_glib/arrow-glib/basic-data-type.cpp:841]:
```suggestion
```

File: c_glib/arrow-glib/basic-data-type.h [hunk 3]
```diff
@@ -418,6 +418,22 @@ struct _GArrowTemporalDataTypeClass
...
+                         BINARY_VIEW_DATA_TYPE,
+                         GArrowDataType)
+struct _GArrowBinaryViewDataTypeClass
+{
+  GArrowDataTypeClass parent_class;
+};
+
+GARROW_AVAILABLE_IN_19_0
+GArrowBinaryViewDataType *
+garrow_binary_view_data_type_new(void);
```
Comment (kou) [inline c_glib/arrow-glib/basic-data-type.h:435]: Could you move this to https://github.com/apache/arrow/blob/40b2fca4742e2692a917755fd8db2939e10fa02d/c_glib/arrow-glib/basic-data-type.h#L772 ?

File: c_glib/arrow-glib/basic-data-type.cpp (hunk 1 above)
Comment (hiroyuki-sato) [reply c_glib/arrow-glib/basic-data-type.cpp]: Thanks! Fixed.

File: c_glib/arrow-glib/basic-data-type.cpp (hunk 2 above)
Comment (hiroyuki-sato) [reply c_glib/arrow-glib/basic-data-type.cpp]: Thanks Fixed.

File: c_glib/arrow-glib/basic-data-type.h (hunk 3 above)
Comment (hiroyuki-sato) [reply c_glib/arrow-glib/basic-data-type.h]: Thanks. moved.

File: c_glib/arrow-glib/basic-data-type.h [hunk 4]
```diff
@@ -418,6 +418,22 @@ struct _GArrowTemporalDataTypeClass
   GArrowFixedWidthDataTypeClass parent_class;
 };
 
+#define GARROW_TYPE_BINARY_VIEW_DATA_TYPE (garrow_binary_view_data_type_get_type())
```
Comment (hiroyuki-sato) [inline c_glib/arrow-glib/basic-data-type.h]: Should I move this definition to the bottom? (Just in case)

**Review by hiroyuki-sato — COMMENTED** [review]
@kou Thank you for your review. Please take a look when you get a chance.

File: c_glib/arrow-glib/basic-data-type.h (hunk 4 above)
Comment (kou) [reply c_glib/arrow-glib/basic-data-type.h]: Yes.
`#define GARROW_TYPE_...` should be near the related `G_DECLARE_DERIVABLE_TYPE()` and so on.

File: c_glib/arrow-glib/type.h [hunk 5]
```diff
@@ -116,7 +118,10 @@ typedef enum {
   GARROW_TYPE_LARGE_LIST,
   GARROW_TYPE_MONTH_DAY_NANO_INTERVAL,
   GARROW_TYPE_RUN_END_ENCODED,
-  /* TODO: Remove = 43 when we add STRING_VIEW(39)..LARGE_LIST_VIEW(42). */
+  GARROW_TYPE_STRING_VIEW,
```
Comment (kou) [inline c_glib/arrow-glib/type.h:121]: Could you add this when we add support string view data type?

```suggestion
```

File: c_glib/arrow-glib/basic-data-type.h (hunk 4 above)
Comment (hiroyuki-sato) [reply c_glib/arrow-glib/basic-data-type.h]: Thanks. moved to the bottom (inside `G_BEGIN_DECL`/`G_END_DECL`)

File: c_glib/arrow-glib/type.h (hunk 5 above)
Comment (hiroyuki-sato) [reply c_glib/arrow-glib/type.h]: Thanks Removed.

File: c_glib/arrow-glib/type.h [hunk 6]
```diff
@@ -70,6 +70,8 @@ G_BEGIN_DECLS
  * @GARROW_TYPE_LARGE_LIST: A list of some logical data type with 64-bit offsets.
  * @GARROW_TYPE_MONTH_DAY_NANO_INTERVAL: MONTH_DAY_NANO interval in SQL style.
  * @GARROW_TYPE_RUN_END_ENCODED: Run-end encoded data.
+ * @GARROW_TYPE_BINARY_VIEW: Bytes view type with 4-byte prefix and inline small string
+ * optimization.
```
Comment (kou) [inline c_glib/arrow-glib/type.h:74]:
```suggestion
 *   optimization.
```

**Review by kou — APPROVED** [review]
+1

**Pull request merged**